Source/WebKit/UIProcess/WebProcessPool.cpp

```diff
@@ -1809,8 +1810,18 @@ void WebProcessPool::removeProcessFromOriginCacheSet(WebProcessProxy& process)
     });
 }
 
-void WebProcessPool::processForNavigation(WebPageProxy& page, const API::Navigation& navigation, Ref<WebProcessProxy>&& sourceProcess, const URL& sourceURL, ProcessSwapRequestedByClient processSwapRequestedByClient, WebProcessProxy::LockdownMode lockdownMode, const FrameInfoData& frameInfo, Ref<WebsiteDataStore>&& dataStore, CompletionHandler<void(Ref<WebProcessProxy>&&, SuspendedPageProxy*, const String&)>&& completionHandler)
+void WebProcessPool::processForNavigation(WebPageProxy& page, const API::Navigation& navigation, Ref<WebProcessProxy>&& sourceProcess, const URL& sourceURL, ProcessSwapRequestedByClient processSwapRequestedByClient, WebProcessProxy::LockdownMode lockdownMode, const FrameInfoData& frameInfo, Ref<WebsiteDataStore>&& dataStore, CompletionHandler<void(Ref<WebProcessProxy>&&, SuspendedPageProxy*, const String&, bool)>&& completionHandler)
 {
+
+    auto registrableDomain = RegistrableDomain { navigation.currentRequest().url() };
+    WTFLogAlways("Chirag -- finding WebProcessPool::processForNavigation for %s", registrableDomain.string().ascii().data());
+    auto* subframePageProxy = page.subpageFrameProxyForRegistrableDomain(registrableDomain);
```

Comment:
We should probably pass enough parameters here to do this only if site isolation is enabled, and only if this is a navigation of a subframe.